Cabal/src/Distribution/Simple/Errors.hs

```diff
@@ -301,6 +302,7 @@ exceptionCode e = case e of
   NoProgramFound{} -> 7620
   BadVersionDb{} -> 8038
   UnknownVersionDb{} -> 1008
+  CombineObjectFilesRelocatableUnsupported -> 4346
```

Comment:
Could not find any information about how this number is chosen. Found another relocation based exception and chose an unused number nearby.

Reply:
@SuganyaAK, @sclv, indeed we should make sure that this gets documented somehow. Would one of you be willing to do so?

Reply:
I think the relevant discussion is https://github.com/haskell/cabal/issues/8543